File: frontend/src/types/therapist.ts
export interface Approach {
    name: string;
    description?: string;
}

export interface Speciality {
    name: string;
    description?: string;
}

export interface Therapist {
    id: number;
    name: string;
    full_name: string | null;
    pronouns: string | null;
    title: string | null;
    credentials: string | null;
    status: string | null;
    intro: string | null;
    ideal_client: string | null;
    approaches: Approach[] | null;
    rate_min: string | null;
    rate_max: string | null;
    free_consultation: boolean | null;
    practicing_since: string | null;
    languages: string | null;
    services: string[] | null;
    insurance: string[] | null;
    specialities: Speciality[] | null;
    other_techniques: string[] | null;
    other_issues: string[] | null;
    url: string | null;
    image?: string;
    location?: string;
    accepting_clients?: boolean;
    telehealth?: boolean;
    in_person?: boolean;
}

export interface SearchQuery {
    query?: string;  // For backward compatibility
    criteria?: string[];  // New multi-criteria support
    insurance?: string[];
    titles?: string[];
}

export interface Filters {
    insurance: string[];
    titles: string[];
}

File: backend/app/models/therapist.py
from sqlalchemy import Column, Integer, String, Text, Float, ARRAY, Boolean, JSON
from sqlalchemy.dialects.postgresql import ARRAY, JSONB
from app.db.base import Base
from sqlalchemy import text

class Therapist(Base):
    __tablename__ = "therapists"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, nullable=False)
    full_name = Column(String, nullable=True)
    pronouns = Column(String, nullable=True)
    title = Column(String, nullable=True)
    credentials = Column(String, nullable=True)
    status = Column(String, nullable=True)
    intro = Column(String, nullable=True)
    ideal_client = Column(String, nullable=True)
    approaches = Column(ARRAY(JSONB), nullable=True)
    rate_min = Column(String, nullable=True)
    rate_max = Column(String, nullable=True)
    free_consultation = Column(Boolean, nullable=True)
    practicing_since = Column(String, nullable=True)
    languages = Column(String, nullable=True)
    services = Column(ARRAY(String), nullable=True)
    insurance = Column(ARRAY(String), nullable=True)
    specialities = Column(ARRAY(JSONB), nullable=True)
    other_techniques = Column(ARRAY(String), nullable=True)
    other_issues = Column(ARRAY(String), nullable=True)
    url = Column(String, nullable=True)
    image = Column(String, nullable=True)
    telehealth = Column(Boolean, nullable=True)
    in_person = Column(Boolean, nullable=True)
    embedding = Column(ARRAY(Float), nullable=True)

    def __init__(self, **kwargs):
        super().__init__(**kwargs)
        if 'embedding' in kwargs:
            self.embedding = kwargs['embedding']
